order-api/src/main/java/com/smoothstack/order/api/OrderApiController.java

```diff
@@ -43,16 +44,17 @@ public OrderApiController(NativeWebRequest request, OrderService orderService) {
 
     @Override
     @PreAuthorize("hasAuthority('user')")
-    public ResponseEntity<CreateResponse> createOrder(Order order) throws MissingFieldsException, EmptyCartException, OrderTimeException {
+    public ResponseEntity<CreateResponse> createOrder(Order order, Long userId) throws MissingFieldsException, EmptyCartException, OrderTimeException, UserNotFoundException {
        if (!order.checkRequiredFields())
             throw new MissingFieldsException("Missing require fields");
        if (order.getFood().size() == 0)
            throw new EmptyCartException("No items in cart.");
-        return orderService.createOrder(order);
+        return orderService.createOrder(order, userId);
     }
 
-/*    @PutMapping(path = "/order/sample")
-    public ResponseEntity<CreateResponse> addSampleOrder (){
-        return ResponseEntity.ok(orderService.createSampleOrder());
-    }*/
-}
+    @PreAuthorize("permitAll")
+    @GetMapping (path = "/order/user", produces = {"application/json"})
+    public ResponseEntity<List<Order>> getUserOrders (@Valid @RequestParam (value = "userId", required = true) Long userId) throws UserNotFoundException {
```

Comment:
This is not secured? Anybody can request any user's orders. This functionality should only be available to admins.